packages/interaction-kit/src/commands/options/subcommand.ts

```diff
@@ -0,0 +1,86 @@
+import type { APIApplicationCommandSubcommandOption } from "discord-api-types/v10";
+import { ApplicationCommandOptionType } from "discord-api-types/v10";
+import type SlashCommandInteraction from "../../interactions/application-commands/slash-command-interaction";
+import type SlashCommandAutocompleteInteraction from "../../interactions/autocomplete/application-command-autocomplete";
+import type { Autocomplete } from "../../interactions/autocomplete/types";
+import type { InteractionKitCommand } from "../../interfaces";
+import type { AutocompleteCommandArgs, CommandArgs } from "../slash-command";
+import type { BasicOption } from "./option";
+import { Option } from "./option";
+
+export default class Subcommand
+	extends Option
+	implements
+		Omit<
+			InteractionKitCommand<SlashCommandInteraction>,
+			"type" | "serialize" | "equals"
+		>,
+		Autocomplete<SlashCommandAutocompleteInteraction>
+{
+	public readonly type = ApplicationCommandOptionType.Subcommand;
+
+	options: Map<string, BasicOption>;
+	handler: InteractionKitCommand<SlashCommandInteraction>["handler"];
+	autocomplete?:
+		| Autocomplete<SlashCommandAutocompleteInteraction>["autocomplete"];
+
+	constructor({
+		name,
+		description,
+		options,
+		handler,
+		autocomplete,
+	}: CommandArgs | AutocompleteCommandArgs) {
+		super({
+			type: ApplicationCommandOptionType.Subcommand,
+			name,
+			description,
+		});
+
+		// TODO: This is somewhat shared with SlashCommand.
+		// TODO: Validate: 1-32 lowercase character name matching ^[\w-]{1,32}$
```

Comment:
You mean long boye regex we have in -builders, right?

Reply:
Yeah, this will be in a followup tho (copied from slash command)